### PR TITLE
Adds Ability to Search for Nodes in Compared Execution Plans

### PR DESCRIPTION
**Pull request by lewis-sanchez**
This PR adds the ability to search for nodes in each of the execution plans that are being compared.

- Clicking the "Find Node" button brings up the search widget for the primary plan (top or left depending on orientation) 
image
image
- Clicking the "Find Node - Added Plan" button brings up the search widget for the added plan (bottom or right depending on orientation). 
image
image

Searching results in the two plans synchronizing and highlighting node the similar node between the two plans.:
image
image

### CODE REVIEW

File: src/sql/workbench/contrib/executionPlan/browser/media/executionPlan.css [hunk 1]
```diff
@@ -694,6 +694,31 @@ However we always want it to be the width of the container it is resizing.
 	border-bottom: 1px solid var(--separator-border);
 }
 
+/* views created by the action-bar actions */
```
Comment (aasimkhan30) [inline src/sql/workbench/contrib/executionPlan/browser/media/executionPlan.css:697]: Are these css duplicated from the previous control? If yes, can you please add the class names to the older css 

for ex:

class1 {
 css
}

becomes 

class1,
class2 {
css code
}

File: src/sql/workbench/contrib/executionPlan/browser/widgets/nodeSearchWidget.ts [hunk 2]
```diff
@@ -37,24 +37,27 @@ export class NodeSearchWidget extends ExecutionPlanWidgetBase {
...
 	private _usePreviousSearchResult: boolean = false;
 
 	private _actionBar: ActionBar;
 
+	private _secondExecutionPlanDiagram: AzdataGraphView;
+
 	constructor(
 		public readonly planActionView: ExecutionPlanWidgetController,
-		public readonly executionPlanDiagram: AzdataGraphView,
+		private readonly executionPlanDiagram: AzdataGraphView,
```
Comment (aasimkhan30) [inline src/sql/workbench/contrib/executionPlan/browser/widgets/nodeSearchWidget.ts:51]: Generally we add _ to private vars

File: src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts [hunk 3]
```diff
@@ -77,7 +83,7 @@ export class ExecutionPlanComparisonEditorView {
 		bottomPolygon: azdata.executionPlan.ExecutionGraphComparisonResult
 	}> = new Map();
 
-	private get _activeTopPlanDiagram(): AzdataGraphView {
+	public get _activeTopPlanDiagram(): AzdataGraphView {
```
Comment (aasimkhan30) [inline src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts:86]: Remove _

**Review by aasimkhan30 — APPROVED** [review]

**Comment by aasimkhan30** [comment]
Looks good to me. Please check if the behavior is consistent with SSMS.

**Comment by lewis-sanchez** [comment]
@erinstellato-ms, can I get your thoughts on the label I used for the second "Find Node - Added Plan" button?

File: src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts [hunk 4]
```diff
@@ -640,6 +666,35 @@ class PropertiesAction extends Action {
...
+		super(SearchNodeAction.ID, getLabelForAction(), searchIconClassNames);
+		this.enabled = false;
+	}
+
+	public override async run(context: ExecutionPlanComparisonEditorView): Promise<void> {
+		let executionPlan = this._planIdentifier === PlanIdentifier.Added ? context.activeBottomPlanDiagram : context.activeTopPlanDiagram;
+		let widgetController = this._planIdentifier === PlanIdentifier.Added ? context.bottomWidgetController : context.topWidgetController;
+
+		this.telemetryService.sendActionEvent(TelemetryKeys.TelemetryView.ExecutionPlan, TelemetryKeys.TelemetryAction.FindNode);
+		let nodeSearchWidget = context._instantiationService.createInstance(NodeSearchWidget, widgetController, executionPlan);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts:693]: You shouldn't be using services from other objects like this - instead inject it into the constructor of this action

File: src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts [hunk 5]
```diff
@@ -640,6 +666,35 @@ class PropertiesAction extends Action {
...
+
+		super(SearchNodeAction.ID, getLabelForAction(), searchIconClassNames);
+		this.enabled = false;
+	}
+
+	public override async run(context: ExecutionPlanComparisonEditorView): Promise<void> {
+		let executionPlan = this._planIdentifier === PlanIdentifier.Added ? context.activeBottomPlanDiagram : context.activeTopPlanDiagram;
+		let widgetController = this._planIdentifier === PlanIdentifier.Added ? context.bottomWidgetController : context.topWidgetController;
+
+		this.telemetryService.sendActionEvent(TelemetryKeys.TelemetryView.ExecutionPlan, TelemetryKeys.TelemetryAction.FindNode);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts:692]: Whenever you're adding any telemetry events please always @ me so I can verify things look good before it's merged in

**Review by erinstellato-ms — APPROVED** [review]
I can't think of a different way to describe the added plan (for the button), so I think this is good.

File: src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts [hunk 6]
```diff
@@ -108,7 +117,7 @@ export class ExecutionPlanComparisonEditorView {
 
 	constructor(
 		parentContainer: HTMLElement,
-		@IInstantiationService private _instantiationService: IInstantiationService,
+		@IInstantiationService public readonly _instantiationService: IInstantiationService,
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts:120]: This should be made private again

File: src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts [hunk 7]
```diff
@@ -95,7 +104,7 @@ export class ExecutionPlanComparisonEditorView {
 	private _bottomSimilarNode: Map<string, azdata.executionPlan.ExecutionGraphComparisonResult> = new Map();
 	private _latestRequestUuid: string;
 
-	private get _activeBottomPlanDiagram(): AzdataGraphView {
+	public get activeBottomPlanDiagram(): AzdataGraphView {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts:107]: The return type should be `AzdataGraphView | undefined` here (same for the activeTopPlanDiagram)

And make sure you're handling undefined properly in all cases - I see some code below where that isn't being checked.

File: src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts [hunk 8]
```diff
@@ -185,10 +201,14 @@ export class ExecutionPlanComparisonEditorView {
 		this._topPlanDropdown = new SelectBox(['option 1', 'option2'], 'option1', this.contextViewService, this._topPlanDropdownContainer);
 		this._topPlanDropdown.render(this._topPlanDropdownContainer);
 		this._topPlanDropdown.onDidSelect(async (e) => {
-			if (this._activeBottomPlanDiagram) {
-				this._activeBottomPlanDiagram.clearSubtreePolygon();
+			if (this.activeBottomPlanDiagram) {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts:204]: FYI - you can simplify these kind of calls by using optional chaining `this.activeBottomPlanDiagram?.clearSubtreePolygon()`

File: src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts [hunk 9]
```diff
@@ -373,12 +415,13 @@ export class ExecutionPlanComparisonEditorView {
 			this._bottomPlanDropdown.select(preSelectIndex);
 			this._propertiesView.setBottomElement(executionPlanGraphs[0].root);
 			this._addExecutionPlanAction.enabled = false;
+			this._searchNodeActionForAddedPlan.enabled = true;
 		}
 		this.refreshSplitView();
 	}
 
 	private async getSkeletonNodes(): Promise<void> {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts:423]: FYI, right now you have this async but it's not actually doing any async work. You call `withProgress` but then don't await it - did you mean to do that? Currently it kicks off the progress task but then immediately returns (while the async progress task is running in the background)

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**